hmrc-mongo-common/src/main/scala/uk/gov/hmrc/mongo/MongoUtils.scala

```diff
@@ -84,6 +85,69 @@ trait MongoUtils {
                             }
     } yield res
 
+   /**
+     * @return the state of the ttl index.
+     * @param checkType true will additionally check the ttl field type. This is an expensive check
+     * (no index) so may not always be appropriate.
+     */
+   private[mongo] def getTtlState(
+    mongoComponent: MongoComponent,
+    collectionName: String,
+    checkType     : Boolean
+  )(implicit ec: ExecutionContext): Future[Map[String, TtlState]] =
+    for {
+      collection <- Future.successful(mongoComponent.database.getCollection(collectionName))
+      indexes    <- collection.listIndexes().toFuture()
+      res        <- indexes.foldLeft(Future.successful(Map.empty[String, TtlState]))((acc, idx) =>
```

Comment:
The `Future` here feels contrived. Would it be better to leave the future out from this bit and then wrap the whole thing in a `Future.successful` instead? (code might be simpler without the Future.successes)